syntax = "proto3";

package csw_protobuf;

import "scalapb/scalapb.proto";

enum PbUnits {
    option (scalapb.enum_options).type = "csw.params.core.models.Units";
    //default unit
    NoUnits = 0;

    // SI units
    angstrom    = 1;
    arcmin      = 2;
    arcsec      = 3;
    day         = 4;
    degree      = 5;
    elvolt      = 6;
    gram        = 7;
    hour        = 8;
    hertz       = 9;
    joule       = 10;
    kelvin      = 11;
    kilogram    = 12;
    kilometer   = 13;
    liter       = 14;
    meter       = 15;
    marcsec     = 16;
    millimeter  = 17;
    millisecond = 18;
    micron      = 19;
    micrometer  = 20;
    minute      = 21;
    newton      = 22;
    pascal      = 23;
    radian      = 24;
    second      = 25;
    sday        = 26;
    steradian   = 27;
    microarcsec = 28;
    volt        = 29;
    watt        = 30;
    week        = 32;
    year        = 33;

    // CGS units
    coulomb    = 34;
    centimeter = 35;
    erg        = 36;

    // Astropyhsics units
    au        = 37;
    jansky    = 38;
    lightyear = 39;
    mag       = 40;

    // Imperial units
    cal   = 41;
    foot  = 42;
    inch  = 43;
    pound = 44;
    mile  = 45;
    ounce = 46;
    yard  = 47;

    // Others - engineering
    encoder = 48;
    count   = 49;
    pix     = 50;
}
